### PR TITLE
[flutter_tools] cleanup fuchsia tests

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Cleanup fuchsia tests that failed under tester.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 1]
```diff
@@ -533,19 +533,20 @@ void main() {
         throwsA(equals('Failed to copy screenshot from device:\n<error-message>')),
       );
     }, overrides: <Type, Generator>{
-      ProcessManager: () => mockProcessManager,
+      ProcessManager: () => MockProcessManager(),
+      FileSystem: () => MemoryFileSystem.test(),
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:537]: I'm not sure how any of these passed with a missing file system override, but it looked to be using the real file system sometimes

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 2]
```diff
@@ -432,19 +432,18 @@ void main() {
...
-    });
-
-    test('is supported on posix platforms', () {
+    testUsingContext('is supported on posix platforms', () {
       final FuchsiaDevice device = FuchsiaDevice('id', name: 'tester');
       expect(device.supportsScreenshot, true);
-    }, testOn: 'posix');
+    }, overrides: <Type, Generator>{
+      Platform: () => FakePlatform(
+        operatingSystem: 'linux',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:440]: replaced skip with platform override

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 3]
```diff
@@ -458,12 +457,12 @@ void main() {
         () => device.takeScreenshot(globals.fs.file('file.invalid')),
         throwsA(equals('file.invalid must be a .ppm file')),
       );
-    }, testOn: 'posix');
+    });
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:460]: can this be `testWithoutContext`?

**Review by christopherfujino — APPROVED** [review]
LGTM with one nit

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:460]: The fuchsia device still uses context :(

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:460]: LGTM then

**Comment by jonahwilliams** [comment]
Using this to kick the tree

**Pull request merged**